py/desispec/quicklook/qlconfig.py

```diff
@@ -244,95 +268,143 @@ def io_qa(self,qaname):
 
         return (outfile,outfig)
 
+    def expand_config(self):
+        """
+        config: desispec.quicklook.qlconfig.Config object
+        """
+        log.info("Building Full Configuration")
+
+        self.flavor = self.conf["flavor"]
+        self.debuglevel = self.conf["Debuglevel"]
+        self.period = self.conf["Period"]
+        self.timeout = self.conf["Timeout"]
+        self.fiberflatexpid = self.conf["FiberflatExpid"]
+        self.psftype = self.conf["PSFType"]
+
+        #- some global variables:
+        self.rawfile=findfile("raw",night=self.night,expid=self.expid, camera=self.camera, rawdata_dir=self.rawdata_dir,specprod_dir=self.specprod_dir)
+
+        self.fibermap=findfile("fibermap", night=self.night,expid=self.expid,camera=self.camera, rawdata_dir=self.rawdata_dir,specprod_dir=self.specprod_dir)
+ 
+        self.fiberflat=findfile("fiberflat",night=self.night,expid=self.fiberflatexpid,camera=self.camera, rawdata_dir=self.rawdata_dir,specprod_dir=self.specprod_dir) #- TODO: Assuming same night for calibration files (here and psf)
+        
+        self.psf=findfile(self.psftype,night=self.night,expid=self.expid,camera=self.camera, rawdata_dir=self.rawdata_dir,specprod_dir=self.specprod_dir)  
+        
+        outconfig={}
+
+        outconfig['Night'] = self.night
+        outconfig['Flavor'] = self.flavor
+        outconfig['Camera'] = self.camera
+        outconfig['Expid'] = self.expid
+        outconfig['DumpIntermediates'] = self.dumpintermediates
+        outconfig['FiberMap']=self.fibermap
+        outconfig['FiberFlatFile'] = self.fiberflat
+        outconfig['PSFFile'] = self.psf
+        outconfig['Period'] = self.period
+
+        pipeline = []
+        for ii,PA in enumerate(self.palist):
```

Comment:
'PA' is what we have always used to refer to pipeline algorithms.  In QuickLook, the QA algorithms are independent of pipeline algorithm step (code wise) and can be moved around.  It has been useful for us, and the pipeline is built Around a configurable list of PAs each with configurable list of QAs attached.  A modern example is in qlconfig_dark.yaml, but it has been there for a long time.  So we keep separate labels for PA and QA algorithms and steps